frame/babe/src/lib.rs

```diff
@@ -305,12 +305,31 @@ impl<T: Trait> Module<T> {
 		// epoch 0 as having started at the slot of block 1. We want to use
 		// the same randomness and validator set as signalled in the genesis,
 		// so we don't rotate the epoch.
-		now != sp_runtime::traits::One::one() && {
+		now != One::one() && {
 			let diff = CurrentSlot::get().saturating_sub(Self::current_epoch_start());
 			diff >= T::EpochDuration::get()
 		}
 	}
 
+	/// Return the _best guess_ block number, at which the next epoch change is predicted to happen.
+	///
+	/// In other word, this is only accurate if no slots are missed. Given missed slots, the slot
+	/// number will grow while the block number will not. Hence, the result can be interpreted as an
+	/// upper bound.
+	///
+	/// -------------- IMPORTANT NOTE --------------
```

Comment:
This note probably deserve a test, to assert should_epoch_change(next_epoch_change). Or if the logic get uncoupled between both function this could be also fine.